cpp/tests/experimental/spatial/polygon_bounding_boxes_test.cu

```diff
@@ -0,0 +1,146 @@
+                                                      poly_offsets.end(),
+                                                      ring_offsets.begin(),
+                                                      ring_offsets.end(),
+                                                      vertices.begin(),
+                                                      vertices.end(),
+                                                      bboxes_begin);
+
+  EXPECT_EQ(std::distance(bboxes_begin, bboxes_end), 0);
+}
+
+TYPED_TEST(PolygonBoundingBoxTest, test_one)
+{
+  using T = TypeParam;
+
+  auto poly_offsets = make_device_vector<int32_t>({0});
+  auto ring_offsets = make_device_vector<int32_t>({0});
+  auto vertices     = make_device_vector<vec_2d<T>>(
+    {{2.488450, 5.856625}, {1.333584, 5.008840}, {3.460720, 4.586599}, {2.488450, 5.856625}});
+
+  auto bbox_min = rmm::device_vector<cuspatial::vec_2d<T>>(poly_offsets.size());
+  auto bbox_max = rmm::device_vector<cuspatial::vec_2d<T>>(poly_offsets.size());
+
+  auto bboxes_begin = thrust::make_zip_iterator(bbox_min.begin(), bbox_max.begin());
+
+  auto bboxes_end = cuspatial::polygon_bounding_boxes(poly_offsets.begin(),
+                                                      poly_offsets.end(),
+                                                      ring_offsets.begin(),
+                                                      ring_offsets.end(),
+                                                      vertices.begin(),
+                                                      vertices.end(),
+                                                      bboxes_begin);
+
+  EXPECT_EQ(std::distance(bboxes_begin, bboxes_end), 1);
+
+  auto bbox_min_expected = make_device_vector<vec_2d<T>>({{1.333584, 4.586599}});
+  auto bbox_max_expected = make_device_vector<vec_2d<T>>({{3.460720, 5.856625}});
+  CUSPATIAL_EXPECT_VECTORS_EQUIVALENT(bbox_min, bbox_min_expected);
+  CUSPATIAL_EXPECT_VECTORS_EQUIVALENT(bbox_max, bbox_max_expected);
+}
+
+TYPED_TEST(PolygonBoundingBoxTest, test_small)
+{
+  using T = TypeParam;
+
+  auto poly_offsets = make_device_vector<int32_t>({0, 1, 2, 3});
+  auto ring_offsets = make_device_vector<int32_t>({0, 4, 9, 13});
+  auto vertices     = make_device_vector<vec_2d<T>>({// ring 1 (closed)
+                                                 {2.488450, 5.856625},
+                                                 {1.333584, 5.008840},
+                                                 {3.460720, 4.586599},
+                                                 {2.488450, 5.856625},
+                                                 // ring 2 (open)
+                                                 {5.039823, 4.229242},
+                                                 {5.561707, 1.825073},
+                                                 {7.103516, 1.503906},
+                                                 {7.190674, 4.025879},
+                                                 {5.998939, 5.653384},
```

Comment:
It's funny that in cuspatial, the tests are assuming both kind of ring formats.

Reply:
I figured for APIs where we don't care, I can test both.